src/TraceEvent/EventPipe/EventPipeEventSourceFactory.cs

```diff
@@ -0,0 +1,57 @@
+﻿using FastSerialization;
```

Comment:
There is an expected pattern to using the serialization library that you are not following.   Basically things you want to deserialize you add the IFastSerializable interface.  THis gives you two methods to implement (ToStream and FromStream).  YOu put all the decoding logic into FromStream.  SInce we never create a EventPipeEventSource you can simply throw NotImplemented for ToStream.   

This is what EventPipeEventSource should do.  Then you don't need to have a factory, since you can say 

deserializer.GetEntryObject();
 
And it gets the 'first' object (which is represented by your whole file).   

Also you don't need an explicit V1 class for the EventPipeEventSource.   The idea is that the FromStream should remember the version number if needed (it does not in this case), and adapt what it does accordingly (that is users should not care about versions, by the time FromStream is done it is deserialized (and looks like the latest version).   



TraceLog.InitializeFromFile has an example, but you don't need the 'RegisterFactory' if the type has a default constructor that takes no arguments.